src/main/java/com/dev/cinema/annotation/FieldsValueMatchValidator.java

```diff
@@ -0,0 +1,23 @@
+package com.dev.cinema.annotation;
+
+import java.util.Objects;
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+import org.springframework.beans.BeanWrapperImpl;
+
+public class FieldsValueMatchValidator implements ConstraintValidator<FieldsValueMatch, Object> {
+    private String filed;
+    private String fieldMatch;
+
+    public void initialize(FieldsValueMatch constraint) {
+        this.filed = constraint.field();
+        this.fieldMatch = constraint.fieldMatch();
+    }
+
+    @Override
+    public boolean isValid(Object value, ConstraintValidatorContext constraintValidatorContext) {
```

Comment:
if you want you can use UserDto instead of Object value

Reply:
Yes, but we're better move from abstraction to realization